examples/evm-to-evm-fungible-transfer/src/transfer.ts

```diff
@@ -29,7 +29,7 @@ export async function erc20Transfer(): Promise<void> {
   const params = {
     source: SEPOLIA_CHAIN_ID,
     destination: HOLESKY_CHAIN_ID,
-    sourceNetworkProvider: web3Provider,
+    sourceNetworkProvider: web3Provider as unknown as Eip1193Provider,
```

Comment:
I think it's better to add support for Web3Provider to sourceNetworkProvide as inside implementation we are dealing mostly with Web3Provider. Eip1193Provider is the more abstract interface, so instead of typecasting we can accept both options Web3Provider | Eip1193Provider

Reply:
@Lykhoyda we are creating instance of `Web3Provider` within SDK but accepting a more abstract implementation like `Eip1193Provider`.

Accepting Web3Provider would mean adding some logic to differentiate between a `Web3Provider` and an `Eip1193Provider` which isn't really needed

Examples create a `web3.js` `HttpProvider`, it is compatible with ethers therefore it works well with the `new Web3Provider` inside SDK just like `Eip1193Provider` just that there is an issue with web3 js types